#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/toys/protos/RubberDuck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/balls/protos/FifaSoccerBall.proto"
EXTERNPROTO "../protos/MobilePhone.proto"
EXTERNPROTO "../protos/Wallet.proto"
EXTERNPROTO "../protos/Keys.proto"
EXTERNPROTO "../protos/Jewelry.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.39383173961958734 -0.3967954771851758 0.8291259917247787 4.506379928725676
  position 1.0540807126546528 9.680379816052895 10.649541194343396
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 10 10
}
Keys {
  translation 2 0 0
}
Jewelry {
  translation -2 -1 0
}
Wallet {
  translation -2 2 0
}
MobilePhone {
  translation 1 2 0
}
DEF rover Robot {
  translation 0 3 0.0906677
  rotation 0.0011703311872850734 0.0007742392362803651 -0.9999990154387738 1.5708043867199883
  children [
    Camera {
      translation 0.29 -0.1 0.1
      children [
        DEF cambox Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.1 0.1 0.1
          }
        }
      ]
      name "CAM"
      boundingObject USE cambox
      physics Physics {
      }
      width 640
      height 480
      recognition Recognition {
      }
    }
    DEF ds_left DistanceSensor {
      translation 0.36 0.1 0
      rotation 0 0 1 0.3927
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 1 0.196078 0.0901961
          }
          geometry Box {
            size 0.025 0.025 0.025
          }
        }
      ]
      name "ds_left"
      boundingObject USE sensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.4 1000 0
      ]
      numberOfRays 2
      resolution 1024
    }
    DEF ds_right DistanceSensor {
      translation 0.36 -0.1 0
      rotation 0 0 1 -0.3927
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 1 0.196078 0.0901961
          }
          geometry Box {
            size 0.025 0.025 0.025
          }
        }
      ]
      name "ds_right"
      boundingObject USE sensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.4 1000 0
      ]
      numberOfRays 2
    }
    DEF body Shape {
      appearance PBRAppearance {
        baseColor 0.3 0.3 0.3
        metalness 0
      }
      geometry Box {
        size 0.7 0.4 0.1
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 7.935322489203051
        axis 0 1 0
        anchor 0.175 0.225 0
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.175 0.225 0
        rotation -0.5460537025564275 -0.5923778316264852 0.5923798262280038 4.141206573225286
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0.6 0.552941 0.47451
              metalness 0
            }
            geometry Cylinder {
              height 0.05
              radius 0.1
            }
          }
        ]
        boundingObject USE wheel
        physics DEF Physics_wh Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 7.935321481607252
        axis 0 1 0
        anchor 0.175 -0.225 0
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation 0.175 -0.225 0
        rotation -0.5460540899826991 -0.5923776530622914 0.592379647663201 4.141207170102611
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0.6 0.552941 0.47451
              metalness 0
            }
            geometry Cylinder {
              height 0.05
              radius 0.1
            }
          }
        ]
        name "solid(1)"
        boundingObject USE wheel
        physics DEF Physics_wh Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 7.935322538214223
        axis 0 1 0
        anchor -0.175 0.225 0
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation -0.175 0.225 0
        rotation -0.5460536837113541 -0.5923778403121489 0.5923798349136934 4.141206544192158
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0.6 0.552941 0.47451
              metalness 0
            }
            geometry Cylinder {
              height 0.05
              radius 0.1
            }
          }
        ]
        name "solid(2)"
        boundingObject USE wheel
        physics DEF Physics_wh Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 7.935321468013051
        axis 0 1 0
        anchor -0.175 -0.225 0
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.175 -0.225 0
        rotation -0.5460540952097441 -0.592377650653153 0.5923796452540538 4.141207178155515
        children [
          DEF wheel Shape {
            appearance PBRAppearance {
              baseColor 0.6 0.552941 0.47451
              metalness 0
            }
            geometry Cylinder {
              height 0.05
              radius 0.1
            }
          }
        ]
        name "solid(3)"
        boundingObject USE wheel
        physics DEF Physics_wh Physics {
        }
      }
    }
    DEF LIDAR_support Shape {
      appearance PBRAppearance {
      }
      geometry Cylinder {
        height 0.5
        radius 0.05
      }
    }
    Lidar {
      translation 0 0 0.24
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Cylinder {
            height 0.01
            radius 0.05
          }
        }
      ]
      name "LIDAR"
      numberOfLayers 5
    }
    DEF IMU InertialUnit {
      translation -0.2 0 0.1
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 1
          }
          geometry Box {
            size 0.1 0.1 0.1
          }
        }
      ]
      name "IMU"
    }
  ]
  boundingObject USE body
  physics Physics {
  }
  controller "rover_open_detect"
}
RubberDuck {
  translation -5.399215832629223e-08 4.6205977958328815e-15 0.05988432667514184
  rotation -8.564273749738891e-08 -0.9999999999999962 -3.8829623230445915e-12 8.855181694219643e-07
  scale 3
}
FifaSoccerBall {
  translation -3.6 1 0.11182507193024632
}
